src/pyEQL/functions.py

```diff
@@ -53,10 +64,16 @@ def gibbs_mix(solution1: Solution, solution2: Solution):
     term_list = {concentrate: 0, dilute: 0, blend: 0}
 
     # calculate the entropy change and number of moles solute for each solution
-    for solution in term_list:
-        for solute in solution.components:
-            if solution.get_amount(solute, "fraction") != 0:
-                term_list[solution] += solution.get_amount(solute, "mol") * np.log(solution.get_activity(solute))
+    if activity_correction is True:
+        for solution in term_list:
+            for solute in solution.components:
+                if solution.get_amount(solute, "fraction") != 0:
+                    term_list[solution] += solution.get_amount(solute, "mol") * np.log(solution.get_activity(solute))
+    else:
+        for solution in term_list:
+            for solute in solution.components:
+                if solution.get_amount(solute, "fraction") != 0:
+                    term_list[solution] += solution.get_amount(solute, "mol") * np.log(solution.get_amount(solute, "fraction"))
```

Comment:
To reduce redundancy, can you please consolidate this logic so that the `if` statement only affects the lines `term_list[solution]...` ? Because the first 3 lines in each branch are the same.

Reducing redundant code is always helpful for increasing robustness and reliability, because it minimizes the chances for someone to change something, say, in the first branch and forget to change it in the 2nd branch later on.

Reply:
Revised it. Thank you for checking it out